mcorpc/authz_rego.go

```diff
@@ -0,0 +1,131 @@
+package mcorpc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/choria-io/go-choria/choria"
+	"github.com/choria-io/go-config"
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/rego"
+	"github.com/open-policy-agent/opa/topdown"
+	"github.com/sirupsen/logrus"
+)
+
+type regoPolicy struct {
+	cfg   *config.Config
+	req   *Request
+	agent *Agent
+	log   *logrus.Entry
+}
+
+func regoPolicyAuthorize(req *Request, agent *Agent, log *logrus.Entry) (bool, error) {
+	logger := log.WithFields(logrus.Fields{
+		"authorizer": "regoPolicy",
+		"agent":      agent.Name(),
+		"request":    req.RequestID,
+	})
+
+	authz := &regoPolicy{
+		cfg:   agent.Config,
+		req:   req,
+		agent: agent,
+		log:   logger,
+	}
+
+	return authz.authorize()
+}
+
+func (r *regoPolicy) authorize() (bool, error) {
+	policyFile, err := r.lookupPolicyFile()
+	if err != nil {
+		return false, err
+	}
+
+	pol, err := ioutil.ReadFile(policyFile)
+	if err != nil {
+		return false, err
+	}
+
+	module := string(pol)
+
+	buf := topdown.NewBufferTracer()
+
+	options := []func(*rego.Rego){
+		rego.Query("data.choria.mcorpc.authpolicy.allow"),
+		rego.Module("//choria/mcorpc/policies/"+r.agent.meta.Name, module),
+		rego.Input(r.regoInputs()),
+	}
+
+	if r.log.Logger.GetLevel() == logrus.TraceLevel {
```

Comment:
We dont use trace level in Choria so this would be a noop more or less always, I go for debug as highest level